### ISSUE TITLE
module ps_emailalert v2.4.1 - notifications are not translatable

### ISSUE THREAD

**Issue by HartLarsson** [PrestaShop/PrestaShop#32414]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

if you try to translate these actions from BO you will not find any field:

`'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('You already have set an alert for this product.', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('Request notification registered', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),`

in BO when you try to translate these error/notifications from file actions.php the field are missing:

![image](https://user-images.githubusercontent.com/16102074/235711903-a6dee6cf-7492-4207-b168-cedff5445ebb.png)

### Expected behavior

going to translate the module, the errors and notification fields must appear under Shop so you can translate all the notifications/error messages

### Steps to reproduce

1) install the  module ps_emailalert
2) find a product with zero availability
3) try to reproduce an error are all in english

### PrestaShop version(s) where the bug happened

1.7.8.9

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

ps_emailalerts v2.4.1

**Comment by Hlavtox** [comment]
@HartLarsson This the same issue as with all other modules.

You can only translate strings that are bundled with the current version of the core. :(

**Comment by HartLarsson** [comment]
> @HartLarsson This the same issue as with all other modules.
> 
> You can only translate strings that are bundled with the current version of the core. :(

well other Presta module are translatable.
Witch are the other module you mentioned?

**Comment by JavierEB** [comment]
https://github.com/PrestaShop/ps_emailalerts/commit/3145d0c9beef7576bf056b22ac63a125e0c063ac#diff-3c4b75267556212a75cf0e3b0e8b58f7e1ad5c2804d31940b0612cb980a0f723  @jolelievre @lartist @Hlavtox  Yes, I also checked it and it doesn't work, let's see if you can verify it, thanks 

![imagen](https://user-images.githubusercontent.com/5004396/236932902-3c32331c-929c-463c-b658-2d5c70db1a9e.png)

**Comment by hibatallahAouadni** [comment]
Hello everyone,

We are aware of this issue, it is already in our debug backlog. This issue is a duplicate of #14183. 
To be informed when it's fixed, please subscribe to the issue mentioned above.

Thanks!

**Comment by hibatallahAouadni** [comment]
Duplicate of #14183

**Comment by HartLarsson** [comment]
> Duplicate of #14183

is not a duplicate! the other report [#14183](https://github.com/PrestaShop/PrestaShop/issues/14183) is about Email translation (we can edit manually the html in the while the fix coming) the problem are the messages that there is no way to translate!

```
'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('You already have set an alert for this product.', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('Request notification registered', [], 'Modules.Emailalerts.Shop'),

'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),`
```

**Comment by JavierEB** [comment]
> > Duplicate of #14183
> 
> is not a duplicate! the other report [#14183](https://github.com/PrestaShop/PrestaShop/issues/14183) is about Email translation (we can edit manually the html in the while the fix coming) the problem are the messages that there is no way to translate!
> 
> ```
> 'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),
> 
> 'message' => $this->trans('You already have set an alert for this product.', [], 'Modules.Emailalerts.Shop'),
> 
> 'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),
> 
> 'message' => $this->trans('Request notification registered', [], 'Modules.Emailalerts.Shop'),
> 
> 'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),`
> ```

Yes, indeed, these messages are the ones that cannot be translated. The report is not duplicated... Thanks for the clarification

**Comment by daresh** [comment]
I think it should be reopened.

**Comment by JavierEB** [comment]
> I think it should be reopened.

+1

**Comment by daresh** [comment]
@kpodemski @khouloudbelguith can you guys help here? It would be good to keep this bug in focus :)

**Comment by camlafit** [comment]
Hello

I've just got this trouble also. I've tried dev version.

**Comment by emilleszczak2** [comment]
When it's going to be fixed?

**Comment by kpodemski** [comment]
cc @hibatallahAouadni - indeed, it's not a duplicate of a mentioned issue

**Comment by SmartPhoneLover** [comment]
Any progress on this?
Or maybe someone just did a fork and implemented translations for the rest of the strings could share the code?

**Comment by ChillCode** [comment]
Have you checked with PS 8.1.5? I can translate those fields and the emails are sent in the chosen language.

![alert](https://github.com/PrestaShop/PrestaShop/assets/9787055/901c40d7-e4df-4f5b-b69c-fed062bd397f)

**Comment by dom99it** [comment]
Hello there, any news on the topic? I have the same issue

**Comment by sebster-bstok** [comment]
> cc [@hibatallahAouadni](https://github.com/hibatallahAouadni) - indeed, it's not a duplicate of a mentioned issue

Is there any information? I have the same problem that alerts cannot be translated in this module.

**Comment by kockalabda** [comment]
> > cc [@hibatallahAouadni](https://github.com/hibatallahAouadni) - indeed, it's not a duplicate of a mentioned issue
> 
> Is there any information? I have the same problem that alerts cannot be translated in this module.

Of course, this issue still hasn't been fixed. If your website is not multilingual, you can translate the message directly in the file:

File Path:
/modules/ps_emailalerts/controllers/front/actions.php

Change this;
'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),
to: 'message' => 'YOUR TRANSLATED TEXT',

'message' => $this->trans('You already have set an alert for this product.', [], 'Modules.Emailalerts.Shop'),
to: 'message' => 'YOUR TRANSLATED TEXT',

'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),
to: 'message' => 'YOUR TRANSLATED TEXT',

'message' => $this->trans('Request notification registered', [], 'Modules.Emailalerts.Shop'),
to: 'message' => 'YOUR TRANSLATED TEXT',

'message' => $this->trans('Your email address is invalid.', [], 'Modules.Emailalerts.Shop'),`
to: 'message' => 'YOUR TRANSLATED TEXT',


After making these changes, clear the cache:

From the admin panel: Advanced Parameters > Performance > Clear Cache.
Or manually delete the following folders:
/var/cache/prod/
/var/cache/dev/.

**Comment by kaWa-stack** [comment]
If you want to translate in French or in English, try to modify function processAdd in File Path: 
/modules/ps_emailalerts/controllers/front/actions.php 

like : 
public function processAdd()
{
    $context = Context::getContext();
    $current_language = $context->language->iso_code; // Retrieves the current language (e.g. ‘fr’ or ‘en’)

    // Definition of messages in both languages
    $messages = [
        'invalid_email' => [
            'fr' => 'Votre adresse e-mail est invalide',
            'en' => 'Your e-mail address is invalid',
        ],
        'alert_exists' => [
            'fr' => 'Vous avez déjà une alerte pour ce produit',
            'en' => 'You already have an alert for this product',
        ],
        'notification_registered' => [
            'fr' => 'Demande de notification enregistrée',
            'en' => 'Request notification registered',
        ],
    ];

    // Message selection in the current language, or fallback in English
    $lang = in_array($current_language, ['fr', 'en']) ? $current_language : 'en';

    if ($context->customer->isLogged()) {
        $id_customer = (int) $context->customer->id;
        $customer = new Customer($id_customer);
        $customer_email = (string) $customer->email;
    } elseif (Validate::isEmail((string) Tools::getValue('customer_email'))) {
        $customer_email = (string) Tools::getValue('customer_email');
        $customer = $context->customer->getByEmail($customer_email);
        $id_customer = (isset($customer->id) && ($customer->id != null)) ? (int) $customer->id : null;
    } else {
        exit(json_encode(
            [
                'error' => true,
                'message' => $messages['invalid_email'][$lang],
            ]
        ));
    }

    $id_product = (int) Tools::getValue('id_product');
    $id_product_attribute = (int) Tools::getValue('id_product_attribute');
    $id_shop = (int) $context->shop->id;
    $id_lang = (int) $context->language->id;
    $product = new Product($id_product, false, $id_lang, $id_shop, $context);

    $mail_alert = MailAlert::customerHasNotification($id_customer, $id_product, $id_product_attribute, $id_shop, null, $customer_email);

    if ($mail_alert) {
        exit(json_encode(
            [
                'error' => true,
                'message' => $messages['alert_exists'][$lang],
            ]
        ));
    } elseif (!Validate::isLoadedObject($product)) {
        exit(json_encode(
            [
                'error' => true,
                'message' => $messages['invalid_email'][$lang],
            ]
        ));
    }

    $mail_alert = new MailAlert();

    $mail_alert->id_customer = (int) $id_customer;
    $mail_alert->customer_email = (string) $customer_email;
    $mail_alert->id_product = (int) $id_product;
    $mail_alert->id_product_attribute = (int) $id_product_attribute;
    $mail_alert->id_shop = (int) $id_shop;
    $mail_alert->id_lang = (int) $id_lang;

    if ($mail_alert->add() !== false) {
        exit(json_encode(
            [
                'error' => false,
                'message' => $messages['notification_registered'][$lang],
            ]
        ));
    }

    exit(json_encode(
        [
            'error' => true,
            'message' => $messages['invalid_email'][$lang],
        ]
    ));
}

**Comment by artemevsin** [comment]
I've tried to remove/rename these files and prestashop loads translations from templates because of this
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/Core/Translation/Storage/Provider/ModuleCatalogueLayersProvider.php#L170

Steps to fix:
1/ delete these files:
```
prestashop/app/Resources/translations/default/ModulesEmailalertsShop.xlf
prestashop/app/Resources/translations/default/ModulesEmailalertsAdmin.xlf
```
2/ clear cache by removing 
```
prestashop/var/cache/prod
prestashop/var/cache/dev
```

**Comment by boo-code** [comment]
This no longer reproduces on 9.2. The module is `ps_emailalerts` 3.0.1 there, against 2.4.1 in the report.

All three strings @HartLarsson listed are in the shipped catalogue, `translations/default/ModulesEmailalertsShop.xlf`:

```
Your email address is invalid.                    PRESENT
You already have set an alert for this product.   PRESENT
Request notification registered                   PRESENT
```

and the translator resolves them in the right domain — `php bin/console debug:translation en --domain=Modules.Emailalerts.Shop`:

```
missing   Modules.Emailalerts.Shop   Your email address is invalid.
missing   Modules.Emailalerts.Shop   You already have set an alert for this product.
missing   Modules.Emailalerts.Shop   Request notification registered
missing   Modules.Emailalerts.Shop   My alerts
```

`missing` there only means no `en` translation is installed, which is expected since English is the source language; the point is that the ids exist in the domain, which is what was not the case when this was reported. The catalogue now holds 18 entries for that domain, and the module ships no catalogue of its own, so core's default is what the back office reads.

To be precise about what I checked: this is verified at the catalogue and translator level rather than by opening the translation page, so if anyone still sees an empty field on 9.2 it would be worth a fresh screenshot, because it would then be a different problem from the one reported here.

**On the duplicate call** — @HartLarsson was right to push back. #14183 is about editing the *email templates* of this module and is still open; this one was about `Modules.Emailalerts.Shop` strings in `controllers/front/actions.php` not reaching the catalogue. Different layer, different fix. Worth keeping them apart if #14183 is triaged later, since closing this one as a duplicate would have hidden a problem that has since been fixed on its own.
